exchanges/kraken/kraken.go

```diff
@@ -1027,3 +1029,41 @@ func (k *Kraken) GetCryptoDepositAddress(method, code string) (string, error) {
 
 	return "", errors.New("no addresses returned")
 }
+
+// WithdrawStatus gets the status of recent withdrawals
+func (k *Kraken) WithdrawStatus(currency, method string) ([]WithdrawStatusResponse, error) {
+	var response struct {
+		Error  []string                 `json:"error"`
+		Result []WithdrawStatusResponse `json:"result"`
+	}
+
+	params := url.Values{}
+	params.Set("asset ", currency)
+	if len(method) != 0 {
+		params.Set("method", method)
+	}
+
+	if err := k.SendAuthenticatedHTTPRequest(krakenWithdrawStatus, params, &response); err != nil {
+		return response.Result, err
+	}
+
+	return response.Result, GetError(response.Error)
+}
+
+// WithdrawCancel sends a withdrawal cancelation request
+func (k *Kraken) WithdrawCancel(currency string, amount float64) (WithdrawCancelResponse, error) {
+	var response struct {
+		Error  []string               `json:"error"`
+		Result WithdrawCancelResponse `json:"result"`
+	}
+	params := url.Values{}
+	params.Set("asset ", currency)
+	params.Set("key ", "")
+	params.Set("amount ", fmt.Sprintf("%f", amount))
```

Comment:
amount will need to changed to refid and the response just appears to be a bool if successful

```
URL: https://api.kraken.com/0/private/WithdrawCancel

Input:

aclass = asset class (optional):
    currency (default)
asset = asset being withdrawn
refid = withdrawal reference id
Result: true on success
```